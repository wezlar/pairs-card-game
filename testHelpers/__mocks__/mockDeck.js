const mockDeck = [
  {
    cardIndex: 0,
    hasMatched: false,
    image: 'images/svg-cards/2_of_hearts.svg',
    isFlipped: false,
    name: '2 of hearts',
  },
  {
    cardIndex: 1,
    hasMatched: false,
    image: 'images/svg-cards/3_of_hearts.svg',
    isFlipped: false,
    name: '3 of hearts',
  },
  {
    cardIndex: 2,
    hasMatched: false,
    image: 'images/svg-cards/4_of_hearts.svg',
    isFlipped: false,
    name: '4 of hearts',
  },
  {
    cardIndex: 0,
    hasMatched: false,
    image: 'images/svg-cards/2_of_hearts.svg',
    isFlipped: false,
    name: '2 of hearts',
  },
  {
    cardIndex: 1,
    hasMatched: false,
    image: 'images/svg-cards/3_of_hearts.svg',
    isFlipped: false,
    name: '3 of hearts',
  },
  {
    cardIndex: 2,
    hasMatched: false,
    image: 'images/svg-cards/4_of_hearts.svg',
    isFlipped: false,
    name: '4 of hearts',
  },
];

export default mockDeck;