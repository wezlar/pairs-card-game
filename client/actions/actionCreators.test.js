import {
  cardFlip,
  cardMatched,
  addNewDeck,
  updateLastCard,
  updateScore,
  updateNumberOfCards,
  updateCardsMatched,
} from './actionCreators';

import mockDeck from '../../testHelpers/__mocks__/mockDeck';

describe(`actionCreators.js`, () => {
  describe(`actions`, () => {
    test(`cardFlip()`, () => {
      const payload = 1;
      expect(cardFlip(payload)).toMatchSnapshot();
    });

    test(`cardMatched()`, () => {
      const payload = 1;
      expect(cardMatched(payload)).toMatchSnapshot();
    });

    test(`addNewDeck()`, () => {
      expect(addNewDeck(mockDeck)).toMatchSnapshot();
    });

    test(`updateLastCard()`, () => {
      expect(updateLastCard(1, '2 of hearts')).toMatchSnapshot();
    });

    test(`updateScore()`, () => {
      expect(updateScore(10)).toMatchSnapshot();
    });

    test(`updateNumberOfCards()`, () => {
      expect(updateNumberOfCards(20)).toMatchSnapshot();
    });
    
    test(`updateCardsMatched()`, () => {
      expect(updateCardsMatched(4)).toMatchSnapshot();
    });
  });
});