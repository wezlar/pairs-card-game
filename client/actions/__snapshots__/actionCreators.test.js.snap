// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actionCreators.js actions addNewDeck() 1`] = `
Object {
  "payload": Object {
    "deck": Array [
      Object {
        "cardIndex": 0,
        "hasMatched": false,
        "image": "images/svg-cards/2_of_hearts.svg",
        "isFlipped": false,
        "name": "2 of hearts",
      },
      Object {
        "cardIndex": 1,
        "hasMatched": false,
        "image": "images/svg-cards/3_of_hearts.svg",
        "isFlipped": false,
        "name": "3 of hearts",
      },
      Object {
        "cardIndex": 2,
        "hasMatched": false,
        "image": "images/svg-cards/4_of_hearts.svg",
        "isFlipped": false,
        "name": "4 of hearts",
      },
      Object {
        "cardIndex": 0,
        "hasMatched": false,
        "image": "images/svg-cards/2_of_hearts.svg",
        "isFlipped": false,
        "name": "2 of hearts",
      },
      Object {
        "cardIndex": 1,
        "hasMatched": false,
        "image": "images/svg-cards/3_of_hearts.svg",
        "isFlipped": false,
        "name": "3 of hearts",
      },
      Object {
        "cardIndex": 2,
        "hasMatched": false,
        "image": "images/svg-cards/4_of_hearts.svg",
        "isFlipped": false,
        "name": "4 of hearts",
      },
    ],
  },
  "type": "ADD_NEW_DECK",
}
`;

exports[`actionCreators.js actions cardFlip() 1`] = `
Object {
  "payload": Object {
    "index": 1,
  },
  "type": "CARD_FLIP",
}
`;

exports[`actionCreators.js actions cardMatched() 1`] = `
Object {
  "payload": Object {
    "index": 1,
  },
  "type": "CARD_MATCHED",
}
`;

exports[`actionCreators.js actions updateCardsMatched() 1`] = `
Object {
  "payload": Object {
    "cardsMatched": 4,
  },
  "type": "CARDS_MATCHED",
}
`;

exports[`actionCreators.js actions updateLastCard() 1`] = `
Object {
  "payload": Object {
    "index": 1,
    "name": "2 of hearts",
  },
  "type": "CARD_SELECTED",
}
`;

exports[`actionCreators.js actions updateNumberOfCards() 1`] = `
Object {
  "payload": Object {
    "numberOfCards": 20,
  },
  "type": "UPDATE_NUMBER_OF_CARDS",
}
`;

exports[`actionCreators.js actions updateScore() 1`] = `
Object {
  "payload": Object {
    "score": 10,
  },
  "type": "UPDATE_SCORE",
}
`;
